_javascripts/page-loader.js

```diff
@@ -1,114 +1,68 @@
-    // check and handle links for Serverless versions
-    newLink = "https://docs.openshift.com/serverless/" +
-      newVersion +
-      fileRequested;
-  } else if (['1.8', '1.9'].contains(newVersion)) {
-    // check and handle links for GitOps versions
-    newLink = "https://docs.openshift.com/gitops/" +
-      newVersion +
-      fileRequested;
-  } else {
-    //if distro key is openshift enterprise
-    if (dk == "openshift-enterprise") {
-    newLink = "https://docs.openshift.com/container-platform/" +
-      newVersion +
-      fileRequested;
-    }
-    else if (dk == "openshift-origin"){
-      newLink = "https://docs.okd.io/" +
-      newVersion +
-      fileRequested;
-    }
-  }
+  newLink = `${baseUrl}${newVersion}${fileRequested}`;
 
   // without doing async loads, there is no way to know if the path actually
   // exists - so we will just have to load
   // window.location = newLink;
   // testing async validations
-  $.ajax({
-    type: 'HEAD',
-    url: newLink,
-    success: function() {
-      window.location = newLink;
-    },
-    error: function(jqXHR, exception) {
-      if(jqXHR.status == 404) {
-        list.value = currentVersion;
-        if(confirm("This page doesn't exist in version " + newVersion + ". Click OK to search the " + newVersion + " docs OR Cancel to stay on this page.")) {
-          if (['3.65', '3.66', '3.67', '3.68', '3.69', '3.70', '3.71', '3.72', '3.73', '3.74', '4.0', '4.1'].contains(newVersion)) {
-            window.location = "https://google.com/search?q=site:https://docs.openshift.com/acs/" + newVersion + " " + document.title;}
-          else if (['1.28', '1.29'].contains(newVersion)) {
-            window.location = "https://google.com/search?q=site:https://docs.openshift.com/serverless/" + newVersion + " " + document.title;}
-          else if (['1.8', '1.9'].contains(newVersion)) {
-            window.location = "https://google.com/search?q=site:https://docs.openshift.com/gitops/" + newVersion + " " + document.title;}            
-          else {
-            if (dk == "openshift-enterprise"){
-              window.location = "https://google.com/search?q=site:https://docs.openshift.com/enterprise/" + newVersion + " " + document.title;
-            } else {
-              if (dk == "openshift-origin"){
-                window.location = "https://google.com/search?q=site:https://docs.okd.io/" + newVersion + " " + document.title;
-              }
-            }
-          }
+  fetch(newLink, { method: "HEAD" })
+    .then(async (response) => {
+      if (response.ok) {
+        window.location.href = newLink;
+      } else if (response.status === 404) {
+        const confirmMessage = `This page doesn't exist in version ${newVersion}. Click OK to search the ${newVersion} docs OR Cancel to stay on this page.`;
+        if (await confirmAsync(confirmMessage)) {
+          const searchUrl = `https://google.com/search?q=site:${urlMappings[dk]}${newVersion} ${document.title}`;
```

Comment:
might you need to handle the enterprise special case here too?

Reply:
**praise:** 👏 @nachandr 
I've fixed it in the latest update.